src/dmd/dinterpret.d

```diff
@@ -2102,17 +2102,22 @@ public:
             else
             {
                 e = hasValue(v) ? getValue(v) : null;
-                if (!e && !v.isCTFE() && v.isDataseg())
-                {
-                    error(loc, "static variable `%s` cannot be read at compile time", v.toChars());
-                    return CTFEExp.cantexp;
-                }
                 if (!e)
                 {
-                    assert(!(v._init && v._init.isVoidInitializer()));
-                    // CTFE initiated from inside a function
-                    error(loc, "variable `%s` cannot be read at compile time", v.toChars());
-                    return CTFEExp.cantexp;
+                    if (v.type.size() == 0)
```

Comment:
I am not at all sure this is a good idea.
Also you have removed an assertion.

Reply:
This special case is already here, just look a little above in the [code](https://github.com/dlang/dmd/blob/3db7045051ea8ea7b110ae3bcfbcc005fb2dca05/src/dmd/dinterpret.d#L2082).
Furthermore, the same logic is also used to handle zero-field struct literals, see `visit(StructLiteralExp)` and `visit(DotVarExp)`.

Reply:
oh my ...